ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala

```diff
@@ -442,22 +441,18 @@ trait H2OGridSearchParams extends H2OCommonSupervisedParams {
 
   def getSelectBestModelBy(): String = $(selectBestModelBy)
 
-  def getSelectBestModelDecreasing(): Boolean = $(selectBestModelDecreasing)
+  @DeprecatedMethod("getSelectBestModelOrdering")
+  def getSelectBestModelDecreasing(): Boolean = {
+    val ordering = getSelectBestModelOrdering()
+    ordering == "ASC" // The default value was DESC as well
+  }
+
+  def getSelectBestModelOrdering(): String = $(selectBestModelOrdering)
 
   //
   // Setters
   //
-
-  def setAlgo(value: H2OSupervisedAlgorithm[_, _, _ <: Model.Parameters]): this.type = {
-    val field = PojoUtils.getFieldEvenInherited(value, "parameters")
-    field.setAccessible(true)
-    val algoParams = field.get(value).asInstanceOf[Model.Parameters]
-    if (!H2OGridSearch.SupportedAlgos.isSupportedAlgo(algoParams.algoName())) {
-      throw new IllegalArgumentException(s"Grid Search is not supported for the specified algorithm '$value'. Supported " +
-        s"algorithms are ${H2OGridSearch.SupportedAlgos.allAsString}")
-    }
-    set(gridAlgoParams, algoParams)
-  }
+  def setAlgo(value: H2OSupervisedAlgorithm[_, _, _ <: Model.Parameters]): this.type = set(algo, value)
```

Comment:
Couldn't the output algorithm be any from supported algorithms? Is this information determined during `fit` phase?

Reply:
I moved the checked for the algorithm into the setter, so the user is warned right away if the incorrect algo is being set